### ISSUE TITLE
Routing Issues - API controllers don't work.

### ISSUE THREAD

**Issue by mattjohnsonpint** [johnpapa/HotTowel#2]
From a new project from the HotTowel template vsix, I added new ApiController to it.  It doesn't work.

I'm not sure exactly why, but if I remove the HotTowelRouteConfig.cs, then the API controllers work again.

It looks like all that it was trying to do was to change the default controller.  In RouteConfig.cs, I just changed "Home" to "HotTowel" instead, and it works just fine.

Am I missing something? Why was a custom route needed?

Also - I added the Microsoft.AspNet.WebApi.HelpPage package.  It too failed until I changed the routing as described.

**Comment by johnpapa** [comment]
This is a question and not an issue or bug, so I prefer this go to StackOverflow. i am not sure what you are doing, as the api and view routes are different :)

**Comment by mika76** [comment]
Actually I think the problem is breeze in this case. Breeze changes the route for the ApiController. Look at BreezeWebApiConfig

**Comment by wardbell** [comment]
It is not obviously a "breeze problem". 

First, changing the MVC routing (which he is doing by toying with HotTowelRouteConfig.cs) shouldn't have an affect on a Web API controller regardless of what the BreezeWebApiConfig is doing. 

Second, HT is deliberately designed to favor development with a Breeze Web API controller which has an action-oriented API and therefore requires a different routing preference.

Third, you can combine Breeze and non-Breeze Web API controllers on the same base endpoint path as long as you distinguish between them effectively in your routing. Which brings us to the next point ...

Fourth, before deviating from the recommended path for Breeze Web API controllers, it's worth acquiring some acquaintance with Web API itself. You can't just wing it by throwing controllers around and expect it to work.

**Comment by mika76** [comment]
I did not mean breeze itself is a problem, just his problem in this case.

I bet he's trying to use the WebApi with the default routing that Microsoft has been pushing everywhere and not the action oriented one which is in use by default in HotTowel.

Matt, try using:

``` GlobalConfiguration.Configuration.Routes.MapHttpRoute(
name: "DefaultApi",
routeTemplate: "api/{controller}/{id}",
defaults: new { id = RouteParameter.Optional }
); 
```

in your BreezeWebApiController instead of the `api/{controller}/{action}` one. If it helps good. If not, sorry,  I tried.
